gulpfile.js

```diff
@@ -1,10 +1,140 @@
 var gulp = require('gulp'),
-    connect = require('gulp-connect');
- 
+    connect = require('gulp-connect'),
+    clean = require('gulp-clean'),
+    inject = require('gulp-inject'),
+    uglify = require('gulp-uglify'),
+    minify = require('gulp-minify-css'),
+    flatten = require('gulp-flatten');
+
+var bases = {
+    dev: {
+        root: './',
+        app: './app',
+        libs: './app/libs',
+        assets: './assets',
+        testapi: './testapi'
+    },
+    dist: {
+        root: 'dist/',
+        app: 'dist/app/',
+        libs: 'dist/libs/',
+        assets: 'dist/assets',
+        testapi: 'dist/testapi'
+    }
+};
+
+var srcPaths = {
+    index: [
+        './index.html',
+        './app.js',
+        './main.css',
+        './config.js'
+    ],
+    libs: [
+        './bower_components/angular/angular.js',
+        './bower_components/leaflet/leaflet.*',
+        './bower_components/**/angular-animate.js',
+        './bower_components/**/angular-aria.js',
+        './bower_components/**/angular-leaflet-directive.js',
+        './bower_components/**/angular-material.js',
+        './bower_components/**/angular-material.css',
+        './bower_components/**/angular-ui-router.js',
+        './bower_components/**/ngStorage.js',
+    ],
+    assets: ['./assets/**/*'],
+    components: ['./app/**/*'],
+    testapi: ['./testapi/**/*']
+};
+
+var distPaths = {
+    css: [
+    bases.dist.root + 'assets/**/*.css',
+    bases.dist.root + 'libs/**/*.css',
+    bases.dist.root + 'main.css'
+    ],
+    js: [
+        bases.dist.root + 'assets/**/*.js',
+        bases.dist.root + 'libs/angular.*js',
+        bases.dist.root + 'libs/**/*.js',
+        bases.dist.root + 'app.js',
+        bases.dist.root + 'app/components/**/*.js'
+    ]
+};
+
+function distCopy () {
+    gulp.src(srcPaths.index)
+        .pipe(gulp.dest(bases.dist.root));
+    gulp.src(srcPaths.libs)
+        .pipe(flatten())
+        .pipe(gulp.dest(bases.dist.libs));
+    gulp.src(srcPaths.assets)
+        .pipe(gulp.dest(bases.dist.assets));
+    gulp.src(srcPaths.components)
+        .pipe(gulp.dest(bases.dist.app));
+    gulp.src(srcPaths.testapi)
+        .pipe(gulp.dest(bases.dist.testapi));
+
+}
+
+function injectImports ($envType) {
+    if ($envType === 'dist') {
+        var dist_index = gulp.src(bases.dist.root + 'index.html');
+
+        var dist_includes = gulp.src(distPaths.css.concat(distPaths.js), {read:false});
+
+        // Flush the inject blocks so the real include can work
+        dist_index.pipe(inject(gulp.src('')))
+            .pipe(gulp.dest(bases.dist.root));
+
+        // Include really now
+        dist_index.pipe(inject(dist_includes, {addRootSlash: false, ignorePath: bases.dist.root}))
```

Comment:
Line is too long.<br>Identifier 'dist_index' is not in camel case.<br>Identifier 'dist_includes' is not in camel case.